src/main/java/com/microsoft/spring/data/gremlin/repository/config/EnableGremlinRepository.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.repository.config;
+
+import com.microsoft.spring.data.gremlin.common.Constants;
+import org.springframework.context.annotation.ComponentScan.Filter;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.repository.config.DefaultRepositoryBaseClass;
+import org.springframework.data.repository.query.QueryLookupStrategy;
+
+import java.lang.annotation.*;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Inherited
+@Import(GremlinRepositoryRegistrar.class)
+public @interface EnableGremlinRepository {
+
+    /**
+     * Alias for basePackages.
+     */
+    String[] value() default {};
+
+    /**
+     * Base packages to scan for components with annotations.
+     */
+    String[] basePackages() default {};
+
+    /**
+     * Type-safe version of basePackages
+     */
+    Class<?>[] basePackagesClasses() default {};
+
+    /**
+     * Specifies types for component scan.
+     */
+    Filter[] includeFilters() default {};
+
+    /**
+     * Specifies types for skipping component scan.
+     */
+    Filter[] excludeFilters() default {};
+
+    /**
+     * Specifics the postfix to be used for custom repository implementation class name.
+     */
+    String repositoryImplementationPostFix() default Constants.DEFAULT_REPOSITORY_IMPLEMENT_POSTFIX;
+
+    /**
+     * Configures where to find the Spring Data named Queries properties file.
+     */
+    String namedQueriesLocation() default Constants.DEFAULT_NAMED_QUERIES_LOCATION;
+
+    /**
+     * Specifies the key for lookup strategy.
+     */
+    QueryLookupStrategy.Key queryLookupStragtegy() default QueryLookupStrategy.Key.CREATE_IF_NOT_FOUND;
```

Comment:
can the above 2 been skipped for now, those 2 are used for custom query

Reply:
OK. I will keep an eye on that.